### PR TITLE
[fix](nereids)mv selection should rule out shadow index from schema change

### PR DESCRIPTION
**Pull request by starocean999**
schema change would introduce a shadow index. MV selection should rule out the shadow index.

Issue Number: close #xxx

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java [hunk 1]
```diff
@@ -145,7 +145,8 @@ private List<SyncMaterializationContext> createSyncMvContexts(OlapTable olapTabl
             keyCount += column.isKey() ? 1 : 0;
         }
         for (Map.Entry<String, Long> entry : olapTable.getIndexNameToId().entrySet()) {
-            if (entry.getValue() != baseIndexId) {
+            long indexId = entry.getValue();
+            if (indexId != baseIndexId && !olapTable.isShadowIndex(indexId)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java:149]: add some comment to explain shadowIndex

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**